### PR TITLE
Add project setup and tooling

### PR DESCRIPTION
**Pull request by rexblack**
* Integrate Webpack, React, Typescript, Styled Components, Material UI, Styleguidist, Linting
* Add component example to show up in styleguidist
* Add basic routes and responsive drawer layout
* Provide some basic data example and project structure outline

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Browsers that we support
```
Comment (StrannikAD) [inline .browserslistrc:1]: Out of interest - what sort of browser support are we looking to have with the block explorer. My FE knowledge is limited but I do know that IE does not play well with ES6. On the off chance that anyone does still use IE is there a way where we can have a popup/message to say that chrome/firefox is a preferred browser?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/styles": "^4.9.10",
+    "@openfonts/poppins_all": "^1.44.4",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "react-router-dom": "^5.1.2",
+    "styled-components": "^5.1.0",
+    "whatwg-fetch": "^3.0.0"
+  },
+  "devDependencies": {
```
Comment (StrannikAD) [inline package.json:16]: Any objection to adding nodemon as a devDependency? I've found the automatic reload function to be invaluable when building out applications: https://nodemon.io/

File: src/api/transactions.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export type Transaction = {
```
Comment (StrannikAD) [inline src/api/transactions.ts:1]: Thanks for setting up the structure! I might go down the path of using an await/async for api calls if there are no objections? In terms of better error handling and performance, I think it is preferable. https://blog.pusher.com/promises-async-await/

Comment (rexblack) [reply src/api/transactions.ts:1]: Of course! Since the example is static, I used just `Promise.resolve` here.

File: package.json (hunk 2 above)
Comment (rexblack) [reply package.json:16]: You mean HMR, don't you? It should be already in place, provided by webpack-dev-server.

Comment (StrannikAD) [reply package.json:16]: Excellent - and yeah, sorry, I should have checked. Still thinking back-end dev terms 😆

File: .browserslistrc (hunk 1 above)
Comment (doc-holliday1) [reply .browserslistrc:1]: This can be handled by the bundler config. you can choose to bundle an es5 version as well, if so desired.

**Pull request merged**